### PR TITLE
Update travis config

### PR DESCRIPTION
**Pull request by mimmi20**
I reorganized the configs for travis-ci and phpunit. I dropped the support for HHVM.

I added some properties in the PropertyHolder because they are part of the browscap.ini shipped with php from ApacheFriends. This file was created 2009.

I removed the bootstrap.php because it is not needed now.

This PR replaces PR #182.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,47 +6,48 @@ php:
...
+      env: COVERAGE=yes
+    - php: 7.0
+      env: EXECUTE_CS_CHECK=yes
+    - php: 7.1
+      env: COMPARE=yes
   allow_failures:
-  # may fail due a segmentation fault
     - php: nightly
-  # the tests may require > 4.8GB ram on hhvm, this exeeds the memory limit on the travis-ci boxes so we allow to fail these builds
     - php: hhvm
```
Comment (asgrim) [inline .travis.yml]: This can be removed also if `hhvm` is removed

File: .travis.yml [hunk 2]
```diff
@@ -6,47 +6,48 @@ php:
...
 
-before_script:
+before_install:
   - .travis-scripts/cache-browscap.sh
-  - if [ "`phpenv version-name`" != "hhvm" ]; then echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
-  - if [ "`phpenv version-name`" != "hhvm" ]; then echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
-  - if [ "`phpenv version-name`" == "hhvm" ]; then echo 'opcache.enable=1' >> /etc/hhvm/php.ini; fi
-  - if [ "`phpenv version-name`" == "hhvm" ]; then echo 'opcache.enable_cli=1' >> /etc/hhvm/php.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION != hhvm && $TRAVIS_PHP_VERSION != nightly && $COVERAGE != yes ]]; then phpenv config-rm xdebug.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION != hhvm ]]; then echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
```
Comment (asgrim) [inline .travis.yml]: If `hhvm` removed, this will always be true now :)

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
     </testsuites>
 
     <filter>
-        <whitelist>
+        <whitelist addUncoveredFilesFromWhitelist="true" processUncoveredFilesFromWhitelist="true">
             <directory suffix=".php">src/</directory>
         </whitelist>
     </filter>
+
+    <logging>
```
Comment (asgrim) [inline phpunit.xml.dist]: Don't enable coverage by default IMO; can slow things down a lot

Comment (mimmi20) [inline phpunit.xml.dist]: fixed

File: .travis.yml (hunk 2 above)
Comment (mimmi20) [inline .travis.yml]: fixed

File: .travis.yml (hunk 1 above)
Comment (mimmi20) [inline .travis.yml]: fixed

**Comment by asgrim** [comment]
Ack, sorry @mimmi20 this conflicts due to #181 being merged :(

**Comment by mimmi20** [comment]
@asgrim The conflicts are solved.

**Comment by asgrim** [comment]
Hmm, build has failed, but I can't seem to load the Travis log at the moment. I'll try again later.

**Comment by jaydiablo** [comment]
Seems ok to me: https://travis-ci.org/browscap/browscap-php/builds/206296647

Except for the php-nightly build in the allowed failures section, but that appears to be an issue in phpunit (extended method has a different signature than its parent).

**Comment by asgrim** [comment]
Yes indeed, it seems to have changed now, how odd :D

**Comment by mimmi20** [comment]
Travis was affected by the AWS outage yesterday.

**Pull request merged**